devimint/src/lib.rs

```diff
@@ -50,11 +50,11 @@ where
     let (process_mgr, task_group) = cli::setup(args).await?;
     log_binary_versions().await?;
     let dev_fed = devfed::DevJitFed::new(&process_mgr, false)?;
-    let res = cleanup_on_exit(f(dev_fed.clone(), process_mgr.clone()), task_group).await;
     // workaround https://github.com/tokio-rs/tokio/issues/6463
     // by waiting on all jits to complete, we make it less likely
     // that something is not finished yet and will block in `on_block`
     let _ = dev_fed.finalize(&process_mgr).await;
+    let res = cleanup_on_exit(f(dev_fed.clone(), process_mgr.clone()), task_group).await;
```

Comment:
Why is this needed? IIUC this ruins parallelism by a lot.

Reply:
But without this change, we're running the test (`f()`) before waiting for the federation to be finalized. Isn't that a race condition?

To answer your question, the LNv2 tests fail without this change. I suspect this is because channel opens with LDK are slow (which is why mprocs is taking so long to start) and on `master` we open a channel between CLN and LND first, allowing for a quick time-to-first-channel:

https://github.com/fedimint/fedimint/blob/cae1f0c6a75262bcd3645b8ee77cc3cbc0c2fdb5/devimint/src/devfed.rs#L273-L281

Whereas on this branch, the time-to-all-channels is quicker but the time-to-first-channel is slower since we're waiting for all channels to reach a certain step before proceeding to the next.